### PR TITLE
Draft: decouple kubeseal api

### PR DESCRIPTION
**Pull request by josvazg**
Signed-off-by: Jose Luis Vazquez Gonzalez <josvaz@vmware.com>

**Description of the change**

Draft kubeseal API refactor

### CODE REVIEW

File: cmd/kubeseal/main.go [hunk 1]
```diff
@@ -74,8 +128,136 @@ func initUsualKubectlFlags(overrides *clientcmd.ConfigOverrides, fs *flag.FlagSe
 	clientcmd.BindOverrideFlags(overrides, fs, kflags)
 }
 
+func runKubeseal(w io.Writer, cfg *Config) (err error) {
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:131]:
```suggestion
func runCLI(w io.Writer, cfg *Config) (err error) {
```

File: cmd/kubeseal/main.go [hunk 2]
```diff
@@ -95,9 +277,9 @@ func mainE(w io.Writer, fs *flag.FlagSet, gofs *goflag.FlagSet, args []string) e
 		return nil
 	}
 
-	clientConfig := kubeseal.InitClient(flags.Kubeconfig, &overrides, os.Stdout)
-	cfg := kubeseal.NewConfig(clientConfig, &flags)
-	return kubeseal.Run(w, cfg)
+	clientConfig := initClient(flags.Kubeconfig, &overrides, os.Stdout)
+	cfg := newConfig(clientConfig, &flags)
+	return runKubeseal(w, cfg)
```
Comment (alemorcuq) [inline cmd/kubeseal/main.go:282]:
```suggestion
	return runCLI(w, cfg)
```

File: cmd/kubeseal/main_test.go [hunk 3]
```diff
@@ -21,3 +37,286 @@ func TestVersion(t *testing.T) {
...
+	b, err := os.ReadFile(out.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got, want := string(b), testOldContent; got != want {
+		t.Errorf("got: %q, want: %q", got, want)
+	}
+}
+
+func Test_runKubeseal(t *testing.T) {
```
Comment (alemorcuq) [inline cmd/kubeseal/main_test.go:239]:
```suggestion
func TestRunCLI(t *testing.T) {
```

File: pkg/kubeseal/kubeseal_test.go [hunk 4]
```diff
@@ -129,18 +108,43 @@ func TestParseKey(t *testing.T) {
...
 }
 
-func testConfig(flags *Flags) *Config {
-	clientConfig := testClientConfig()
-	return &Config{
-		flags:          flags,
-		clientConfig:   clientConfig,
-		ctx:            context.Background(),
-		solveNamespace: initNamespaceFuncFromClient(clientConfig),
+/* repeated from main here... STARTs */
```
Comment (alemorcuq) [inline pkg/kubeseal/kubeseal_test.go:111]: What's this comment?

**Review by alemorcuq — CHANGES_REQUESTED** [review]

**Comment by josvazg** [comment]
We need some refactors before this. @alemorcuq will take this over

**Pull request closed without merging**